packages/web/src/common/store/notifications/fetchNotifications.ts

```diff
@@ -46,15 +51,32 @@ export function* fetchNotifications(config: FetchNotificationsParams) {
   const shouldFetchNotificationFromDiscovery =
     useDiscoveryNotifications &&
     discoveryNotificationsGenesisTimestamp &&
-    discoveryNotificationsGenesisTimestamp > Date.parse(timeOffset)
+    Date.parse(timeOffset) > discoveryNotificationsGenesisTimestamp
 
   if (shouldFetchNotificationFromDiscovery) {
     const timestampParam = Math.trunc(Date.parse(timeOffset) / 1000)
+
+    const isRepostOfRepostEnabled = yield* call(
+      getFeatureEnabled,
+      FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS
+    )
+    const isSaveOfRepostEnabled = yield* call(
+      getFeatureEnabled,
+      FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
+    )
+
+    const validTypes = [
+      isRepostOfRepostEnabled ? 'repost_of_repost' : null,
+      isSaveOfRepostEnabled ? 'save_of_repost' : null
+    ].filter(removeNullable)
+
     const discoveryNotifications = yield* call(
       audiusBackendInstance.getDiscoveryNotifications,
       {
         timestamp: timestampParam,
-        groupIdOffset
+        groupIdOffset,
+        limit,
+        validTypes: validTypes.length > 0 ? validTypes : undefined
```

Comment:
i think its fine to just send `validTypes` here empty or not?? is there a problem sending validTypes if its empty?

Reply:
i was noticing if validTypes: [] then it weirdly breaks the api.... can double check on this though